picasso/src/main/java/com/squareup/picasso3/NetworkRequestHandler.java

```diff
@@ -42,30 +42,33 @@
     return (SCHEME_HTTP.equals(scheme) || SCHEME_HTTPS.equals(scheme));
   }
 
-  @Override public Result load(Request request, int networkPolicy) throws IOException {
-    okhttp3.Request callRequest = createRequest(request, networkPolicy);
-    Response response = callFactory.newCall(callRequest).execute();
-    ResponseBody body = response.body();
-
-    if (!response.isSuccessful()) {
-      body.close();
-      throw new ResponseException(response.code(), request.networkPolicy);
-    }
-
-    // Cache response is only null when the response comes fully from the network. Both completely
-    // cached and conditionally cached responses will have a non-null cache response.
-    Picasso.LoadedFrom loadedFrom = response.cacheResponse() == null ? NETWORK : DISK;
+  @Override public void load(Request request, int networkPolicy, Callback callback) {
+    try {
+      okhttp3.Request callRequest = createRequest(request, networkPolicy);
+      Response response = callFactory.newCall(callRequest).execute();
+      if (!response.isSuccessful()) {
+        callback.onError(new ResponseException(response.code(), request.networkPolicy));
```

Comment:
needs a return

Reply:
needs a body.close() ?

Reply:
updated.  re:`close()`, I moved the `response.body()` call lower; `response.code()` doesn't consume the body.